rest/index.py

```diff
@@ -468,6 +481,433 @@ def createIncident():
+
+    ```
+    {
+        "error": Optional[str]
+    }
+    ```
+
+    If an error occurs, no duplicate chains will be updated.  This endpoint
+    does not create a new chain if one does not already exist.
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    try:
+        req = json.loads(request.body.read())
+        request.body.close()
+    except ValueError:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps({"error": "Missing or invalid request body"})
+        return response
+
+    query = {"alert": req.get("alert"), "user": req.get("user")}
+
+    new_ids = req.get("identifiers")
+
+    if any([x is None for x in (query["alert"], query["user"], new_ids)]):
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps(
+            {
+                "error": "Request missing required key `alert`, `user` or "
+                "`identifiers`"
+            }
+        )
+        return response
+
+    chain = dupchains.find_one(query)
+
+    if chain is None:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = json.dumps({"error": "Duplicate chain does not exist"})
+        return response
+
+    modified = dupchains.update_one(
+        query,
+        {
+            "$set": {
+                "identifiers": chain["identifiers"] + new_ids,
+                "modified": datetime.utcnow(),
+            }
+        },
+    ).modified_count
+
+    if modified != 1:
+        response.status = StatusCode.INTERNAL_ERROR
+        response.body = json.dumps({"error": "Failed to update chain"})
+        return response
+
+    response.status = StatusCode.OK
+    response.body = json.dumps({"error": None})
+
+    return response
+
+
+@delete("/alerttriagechain")
+@delete("/alerttriagechain/")
+def delete_duplicate_chain():
+    """Deletes a Duplicate Chain tracking duplicate alerts triggered by the
+    same user.
+
+    Requests are expected to contain the following JSON data:
+
+    ```
+    {
+        "alert": str,
+        "user": str
+    }
+    ```
+
+    Where:
+        * `"alert"` is the label of an alert supported by the triage bot.
+        * `"user"` is the email address of the user the triage bot contacted.
+
+    Responses will contain the following JSON:
+
+    ```
+    {
+        "error": Optional[str]
+    }
+    ```
+
+    In the case that a duplicate chain identified by the request parameters does
+    not exist or an error occurs in deleting it, the `"error"` field will
+    contain a string describing the error.  Otherwise, it is `null`.
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    try:
+        req = json.loads(request.body.read())
```

Comment:
You could use [bottle's native support for POSTed JSON](https://bottlepy.org/docs/dev/api.html#bottle.BaseRequest.json)